crates/bevy_ui/src/lib.rs

```diff
@@ -176,7 +176,7 @@ impl Plugin for UiPlugin {
                 PostUpdate,
                 (
                     CameraUpdateSystem,
-                    UiSystem::Prepare.before(UiSystem::Stack).after(Animation),
+                    UiSystem::Prepare.after(UiSystem::Stack).after(Animation),
                     UiSystem::Layout,
                     UiSystem::PostLayout,
```

Comment:
`UiSystem::Prepare` was explicitly added to be the first ui system set. That way you can reliably do `.before(UiSystem::Prepare)` and your systems will never mess up ui calculations.

It looks like you're using the results of `UiSystem::Stack` in systems in `Prepare` now. I think you need to do a larger rearranging of systems to accommodate this so `Prepare` remains the first set.